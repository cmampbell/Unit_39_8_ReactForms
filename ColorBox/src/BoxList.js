import React, { useState } from "react";
import { v4 as uuid } from "uuid"
import Box from "./Box"
import NewBoxForm from "./NewBoxForm"
import "./BoxList.css"

const BoxList = () => {
    const [boxes, setBoxes] = useState([])

    const addBox = ({ height, width, backgroundColor }) => {
        setBoxes(boxes =>
            [...boxes, { height: +height, width: +width, backgroundColor }]
        )
    }

    const deleteBox = (boxToDelete) => {
        setBoxes(boxes => 
            boxes.filter(box => boxToDelete !== box)
            )
    }

    return (
        <div>
            <NewBoxForm addBox={addBox} />
            {boxes.map(box => <Box boxStyle={box} key={uuid()} deleteBox={deleteBox}/>)}
        </div>
    )
}

export default BoxList;