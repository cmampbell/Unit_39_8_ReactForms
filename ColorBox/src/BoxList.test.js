import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import BoxList from './BoxList';


it('renders without crashing', function() {
    render(<BoxList />);
});

it('matches snapshot', function() {
    const {asFragment} = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});

it('adds box on form submit', function() {
    const {getByLabelText, getByText} = render(<BoxList />);

    fireEvent.change(getByLabelText('Height:'), {target:{value:400}})
    fireEvent.change(getByLabelText('Width:'), {target:{value:400}})
    fireEvent.change(getByLabelText('Color:'), {target:{value:'#a4a4a4'}})


    fireEvent.click(getByText('Create Box'))

    expect(getByText('X')).toBeInTheDocument()
})