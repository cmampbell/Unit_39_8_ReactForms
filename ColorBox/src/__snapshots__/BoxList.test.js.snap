// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        max="500"
        min="1"
        name="width"
        placeholder="1"
        type="number"
        value="1"
      />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        max="500"
        min="1"
        name="height"
        placeholder="1"
        type="number"
        value="1"
      />
      <label
        for="backgroundColor"
      >
        Color:
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        type="color"
        value="#000000"
      />
      <button>
        Create Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
